Microsoft.Toolkit.Uwp.UI/Extensions/Mouse/Mouse.cs

```diff
@@ -0,0 +1,115 @@
+        public static void SetCursor(FrameworkElement element, CoreCursorType value)
+        {
+            element.SetValue(CursorProperty, value);
+        }
+
+        /// <summary>
+        /// Get the current <see cref="CoreCursorType"/>.
+        /// </summary>
+        /// <param name="element">Object where the selector cursor type should be shown.</param>
+        /// <returns>Cursor type set on target element.</returns>
+        public static CoreCursorType GetCursor(FrameworkElement element)
+        {
+            return (CoreCursorType) element.GetValue(CursorProperty);
+        }
+
+        private static void CursorChanged(DependencyObject d, DependencyPropertyChangedEventArgs e)
+        {
+            var element = d as FrameworkElement;
+            if (element == null)
+            {
+                throw new NullReferenceException(nameof(element));
+            }
+
+            var value = (CoreCursorType) e.NewValue;
+
+            // lock ensures CoreCursor creation and event handlers attachment/detachment is atomic
+            lock (s_cursorLock)
+            {
+                if (!s_cursors.ContainsKey(value))
+                {
+                    s_cursors[value] = new CoreCursor(value, 1);
+                }
+
+                // make sure event handlers are not attached twice to element
+                element.PointerEntered -= Element_PointerEntered;
+                element.PointerEntered += Element_PointerEntered;
+                element.PointerExited -= Element_PointerExited;
+                element.PointerExited += Element_PointerExited;
+                element.Unloaded -= ElementOnUnloaded;
+                element.Unloaded += ElementOnUnloaded;
+            }
+        }
+
+        private static void Element_PointerEntered(object sender, PointerRoutedEventArgs e)
+        {
+            CoreCursorType cursor = GetCursor((FrameworkElement) sender);
+            Window.Current.CoreWindow.PointerCursor = s_cursors[cursor];
+        }
+
+        private static void Element_PointerExited(object sender, PointerRoutedEventArgs e)
+        {
+            // when exiting change the cursor to the target Mouse.Cursor value of the new element
+            CoreCursor cursor;
+            if (e.OriginalSource is FrameworkElement newElement)
+            {
+                cursor = s_cursors[GetCursor(newElement)];
+            }
+            else
+            {
+                cursor = s_defaultCursor;
+            }
+
+            Window.Current.CoreWindow.PointerCursor = cursor;
+        }
+
+        private static void ElementOnUnloaded(object sender, RoutedEventArgs routedEventArgs)
+        {
+            // when the element is programatically unloaded, reset the cursor back to default
+            // this is necessary when click triggers immediate change in layout and PointerExited is not called
+            Window.Current.CoreWindow.PointerCursor = s_defaultCursor;
```

Comment:
You need to remove the unloaded control's cursor from the dictionary, because 
1) if the control get loaded again and it's cursor is changed, the dictionary will hold the old cursor of the control.
2) we don't need to maintain unused data in memory.

Reply:
As-is, you don't know if there are more control that need that type of cursor. To remove the cursors, you would need to change the collection Dictionary that track of the number of items. eg: `IDictionary<Cursor, int>`

Reply:
Oh, I thought that this is a dictionary<element,cursor>. 
this way I need to test if a control has a cursor and it's child control has a cursor too, what would happen :)

Reply:
This is a tradeoff decision: either we need to keep dictionary with unique cursor types, or create new CoreCusor whenever it is required. I'd go for the persistent dictionary that would hold usually only about 2-3 CoreCurosr instances in app lifetime, nothing serious.
Maintaining dictionary of cursors per element feels like an overkill to be honest.